assets/app/scripts/directives/quotaUsageChart.js

```diff
@@ -0,0 +1,122 @@
+          if (!donutChartTitle) {
+            Logger.warn("Can't select donut title element");
+            return;
+          }
+
+          var valueAndType = usageWithUnits($scope.total, $scope.type).split(' ');
+          var largeText = valueAndType[0];
+          var smallText = valueAndType[1];
+
+          // Replace donut title content.
+          donutChartTitle.selectAll('*').remove();
+          donutChartTitle.insert('tspan').text(largeText).classed('pod-count donut-title-big-pf', true).attr('dy', 0).attr('x', 0);
+          donutChartTitle.insert('tspan').text(smallText).classed('donut-title-small-pf', true).attr('dy', 20).attr('x', 0);
+        }
+
+        // Adjust size based on legend position.
+        if ($scope.legendPosition === 'right') {
+          $scope.height = 175;
+          $scope.width = 250;
+        } else {
+          $scope.height = 200;
+          $scope.width = 175;
+        }
+
+        // https://github.com/mbostock/d3/wiki/Formatting
+        var percentage = d3.format(".2p");
+
+        // Chart configuration, see http://c3js.org/reference.html
+        $scope.chartID = _.uniqueId('quote-usage-chart-');
+        var config = {
+          type: "donut",
+          bindto: '#' + $scope.chartID,
+          donut: {
+            label: {
+              show: false
+            },
+            width: 10
+          },
+          size: {
+            height: $scope.height,
+            width: $scope.width
+          },
+          legend: {
+            show: true,
+            position: $scope.legendPosition || 'bottom',
+            item: {
+              // Don't hide arcs when clicking the legend.
+              onclick: _.noop
+            }
+          },
+          onrendered: updateCenterText,
+          tooltip: {
+            position: function() {
+              // Position in the top-left to avoid problems with tooltip text wrapping.
+              return { top: 0, left: 0 };
+            },
+            format: {
+              value: function(value) {
+                return percentage(value / usageValue($scope.total)) +
+                  " of " + usageWithUnits($scope.total, $scope.type);
+              }
+            },
+          },
+          data: {
+            type: "donut",
+            // Keep groups in our order.
+            order: null
+          }
+        };
+
+        var chart;
+        var updateChart = function() {
+          var used = usageValue($scope.used) || 0,
+              available = Math.max(usageValue($scope.total) - used, 0),
+              data = {
+                columns: [
+                  ['Used', used],
+                  ['Available', available]
+                ],
+                // https://www.patternfly.org/styles/color-palette/
+                colors: {
+                  // Orange if at quota, blue otherwise
+                  Used: available ? "#0088ce" : "#ec7a08",
```

Comment:
maybe patternfly needs to create a json map of its palette colors ;-)   @rhamilto kidding... sort of..